src/test/java/seedu/application/model/application/UpcomingInterviewPredicateStub.java

```diff
@@ -0,0 +1,35 @@
+package seedu.application.model.application;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * Tests that an {@code Application}'s {@code Interview} is within 1 week from now.
+ * The current date and time reference is pre-configured in this stub and does not follow the system clock.
+ */
+public class UpcomingInterviewPredicateStub implements Predicate<Application> {
```

Comment:
Maybe `UpcomingInterviewPredicateStub` can extend `UpcomingInterviewPredicate`? So that you only need to change one method.

Reply:
thanks for the suggestion! 

I did think about this as well, but decided to follow the textbook definition of a stub, where the stub implemented the parent class. I'm not sure if I can call the stub a stub if I adopted the suggestion instead 😅 

thanks anyways~

![image](https://user-images.githubusercontent.com/26516640/198284446-47a66cf3-2f8b-4522-97d7-d759549fd224.png)

![image](https://user-images.githubusercontent.com/26516640/198284494-3808d581-a637-4111-b207-18a95c503c4a.png)